server/api/rounds.py

```diff
@@ -67,41 +68,100 @@ def validate_round(round: dict, election: Election):
         raise BadRequest("Sample size is required for round 1")
 
 
+class SampleDraw(NamedTuple):
+    # ballot_key: ((jurisdiction name, batch name), ballot_position)
+    ballot_key: Tuple[Tuple[str, str], int]
+    contest_id: str
+    ticket_number: str
+
+
 def sample_ballots(election: Election, round: Round, sample_size: int):
-    # For now, we only support jointly targeted contests, which all must have
-    # the same contest universe. So we can just take any of them to build the
-    # pool of ballots to sample.
-    targeted_contest = next(c for c in election.contests if c.is_targeted)
-
-    # Compute the total number of ballot samples in all rounds leading up to
-    # this one. Note that this corresponds to the number of SampledBallotDraws,
-    # not SampledBallots.
-    num_previously_sampled = (
-        SampledBallotDraw.query.join(Round).filter_by(election_id=election.id).count()
+    # Figure out which contests still need auditing
+    last_round = get_current_round(election)
+    contests_that_havent_met_risk_limit = (
+        [
```

Comment:
should there be guidance on the `round` argument passed here? I'm guessing it's a new round that's later than `last_round`?

Reply:
changed this up to be more clear